File: challenges/7-jokenpo-nerd/python/scush989898/desafio-7.py
game_terms = "tesoura papel pedra lagarto spock tesoura lagarto papel spock pedra tesoura"


test_cases = int(input())
rounds_played = []

for i in range(test_cases):
    inp = str(input()).strip().lower()
    current = inp.split(" ")
    rounds_played.append(current)

for round in rounds_played:
    first = round[0]
    second = round[1]

    if first == second:
        print("empate")
    elif f"{first} {second}" in game_terms:
        print("rajesh")
    elif f"{second} {first}" in game_terms:
        print("sheldon")
    else:
        print("os valores digitados não são válidos")

File: challenges/7-jokenpo-nerd/ruby/jailsonsf/desafio-7.rb
test_cases = gets.to_i

posibilities = {
  pedra: %i[lagarto tesoura],
  papel: %i[pedra spock],
  tesoura: %i[papel lagarto],
  lagarto: %i[spock papel],
  spock: %i[tesoura pedra]
}

def game_result(player_one, player_two, posibilities)
  if player_one == player_two
    'empate'
  elsif posibilities[player_one].include?(player_two)
    'rajesh'
  else
    'sheldon'
  end
end

(1..test_cases).each do |_i|
  player_one, player_two = gets.split.map(&:downcase)

  puts game_result(player_one.to_sym, player_two.to_sym, posibilities)
end
